### PR TITLE
feat: [IOPID-1611] Adopt new DS on CIE not supported modal

### PR DESCRIPTION
**Pull request by Ladirico**
## Short description
Integrate new DS on CIE not supported modal

## List of changes proposed in this pull request
- Integrate new DS on CIE not supported modal
- transform CIE not supported modal in native stack modal

## Test

> [!TIP]
> [Run e2e tests](https://github.com/pagopa/io-app/actions/workflows/test-e2e.yml)

### Accessibility Test
<p>

| 🍏 A11Y iOS 🍏 | 🤖 A11Y Android 🤖 |
| - | - | 
| <video src="https://github.com/pagopa/io-app/assets/83651704/f96ffa6a-900e-47d2-82de-d7b23450d0d1" /> | <video src="https://github.com/pagopa/io-app/assets/83651704/3324a76a-a959-4c5d-bff4-5a5d28e8b1c3" /> |

</p>

## How to test
On _.env_ file set `CIE_LOGIN_WITH_DEV_SERVER_ENABLED=NO`
Run the application and on the landing screen select CIE as login mode

### CODE REVIEW

**Comment by Ladirico** [comment]
@mariateresaventura could you please review the copy? 😁

File: ts/navigation/AuthenticationNavigator.tsx [hunk 1]
```diff
@@ -98,6 +101,28 @@ const AuthenticationStackNavigator = () => (
...
     />
+
+    <Stack.Group
+      screenOptions={{
+        presentation: "modal",
+        headerLeft: () => null,
+        headerTitle: () => null,
+        headerRight: CloseButton,
+        headerStyle: {
+          height: IOVisualCostants.headerHeight,
```
Comment (shadowsheep1) [inline ts/navigation/AuthenticationNavigator.tsx:112]: Can we remove this provided that we set `headerStatusBarHeight: 0`? What do you think?

```suggestion
```

File: ts/components/cie/CieNotSupported.tsx [hunk 2]
```diff
@@ -1,80 +1,101 @@
...
+      primaryActionProps={{
+        label: I18n.t("authentication.landing.cie_unsupported.button"),
+        accessibilityLabel: I18n.t(
+          "authentication.landing.cie_unsupported.button"
+        ),
+        onPress: () => navigation.goBack(),
+        testID: "close-button"
+      }}
+    >
+      <SafeAreaView>
```
Comment (shadowsheep1) [inline ts/components/cie/CieNotSupported.tsx:48]: Isn't the safe area already taken into consideration by the `GradientScrollView`?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: ts/components/cie/CieNotSupported.tsx (hunk 2 above)
Comment (Ladirico) [reply ts/components/cie/CieNotSupported.tsx]: Yess! Thanks!

File: ts/navigation/AuthenticationNavigator.tsx (hunk 1 above)
Comment (Ladirico) [reply ts/navigation/AuthenticationNavigator.tsx:129]: yes, correct! Thanks!

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**